relay-metrics/src/bucket.rs

```diff
@@ -538,7 +537,7 @@ pub struct Bucket {
     ///
     /// Namespaces and units must consist of ASCII characters and match the regular expression
     /// `/\w+/`. The name component of MRIs consist of unicode characters and must match the
-    /// regular expression `/\w[\w\d_-.]+/`. Note that the name must begin with a letter.
+    /// regular expression `/\w[\w\-.]*/`. Note that the name must begin with a letter.
```

Comment:
The validation/normalization implementation in Relay differs from the rules for legal metric names:
- We do not yet enforce the first letter. This will come in as a separate refactor.
- Dashes are intentionally replaced with underscores. Right now, the product wants just underscores, but we still permit SDKs to send dashes for future proofing.

Reply:
> Dashes are intentionally replaced with underscores

Should we add this information to the doc comment here?